requester/requester.go

```diff
@@ -19,7 +26,9 @@ type Requester struct {
 	tracer *tracer
 }
 
-// New returns a Requester configured with specified Options.
+// New returns a Requester initialized with cfg. cfg is assumed valid:
+// it is the caller's responsibility to ensure cfg is valid using
+// cfg.Validate.
 func New(cfg config.Config) *Requester {
 	tracer := newTracer()
 	return &Requester{
```

Comment:
Side not but super important: now that `-1` for number of requests is allowed and expected, calling `make(chan Record, cfg.RunnerOptions.Requests)` can panic because `make` do not accept negative integers.

Reply:
Indeed, noticed the same for `collect`:
https://github.com/benchttp/runner/blob/300bfb6438467cfd86ea38bf58871a6e423e7a6f/requester/report.go#L30-L34

Hence the addition in the docs 3 lines above this comment:
```go
// New returns a Requester initialized with cfg. cfg is assumed valid:
// it is the caller's responsibility to ensure cfg is valid using
// cfg.Validate.
func New(cfg config.Config) *Requester {
```

Reply:
But -1 _is_ a valid config isn't it ? Or did I miss a change ?

Thus we can pass -1 in `make`.

Reply:
Damn why do I keep forgetting that.